modules/Rubrics/src/Visualise.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/*
+Gibbon, Flexible & Open School System
+Copyright (C) 2010, Ross Parker
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace Gibbon\Module\Rubrics;
+
+use Gibbon\Contracts\Database\Connection;
+use Gibbon\session;
+use Gibbon\UI\Chart\Chart;
+
+/**
+ * Attendance display & edit class
+ *
+ * @version 12th Sept 2016
+ * @since   12th Sept 2016
+ * @author  Sandra Kuipers
+ */
+class Visualise
+{
+    /**
+     * Gibbon\Contracts\Database\Connection
+     */
+    protected $pdo;
+
+    /**
+     * Gibbon\session
+     */
+    protected $session;
+
+    protected $page;
+
+    protected $guid;
+
+    protected $gibbonPersonID;
+
+    protected $columns;
+
+    protected $rows;
+
+    protected $cells;
+
+    protected $contexts;
+
+    /**
+     * Constructor
+     *
+     * @version  3rd May 2016
+     * @since    3rd May 2016
+     * @param    Gibbon\session
+     * @param    Gibbon\config
+     * @param    Gibbon\Contracts\Database\Connection
+     * @return   void
+     */
+    public function __construct(\Gibbon\Core $gibbon, Connection $pdo, $page, Int $gibbonPersonID, Array $columns, Array $rows, Array $cells, Array $contexts)
```

Comment:
Looks like the `$pdo` dependency isn't being used by this class. The `$gibbon` dependency appears to only be used for the absoluteURL, so to keep things simple it may work to just pass in the `$absoluteURL` string, and remove the other two class dependencies.

Reply:
Thanks for setting me straight on these. A lot of this work was speculative, and I meant to come back and clean it up...but never did. Fixed now.